aspnetcore/mvc/models/model-binding.md

```diff
@@ -1,146 +1,434 @@
+
+[!code-csharp[](model-binding/samples/2.x/Pages/Instructors/Edit.cshtml.cs?name=snippet_BindProperty&highlight=7-8)]
+
+### [BindProperties] attribute
+
+Available in ASP.NET Core 2.1 and later.  Can be applied to a controller or `PageModel` class to tell model binding to target all public properties of the class:
+
+[!code-csharp[](model-binding/samples/2.x/Pages/Instructors/Create.cshtml.cs?name=snippet_BindProperties&highlight=1-2)]
+
+### Model binding for HTTP GET requests
+
+By default, the `[BindProperty]` and `[BindProperties]` attributes have no effect for HTTP GET requests. Typically, all you need for a GET request is a record ID parameter. The record ID is used to look up the item in the database. Therefore, there is no need to bind a property that holds an instance of the model. In scenarios where you do want properties bound to data from GET requests, set the `SupportsGet` property to `true`:
+
+[!code-csharp[](model-binding/samples/2.x/Pages/Instructors/Index.cshtml.cs?name=snippet_SupportsGet)]
+
+## Sources
+
+By default, model binding gets data in the form of key-value pairs from the following sources in an HTTP request:
+
+1. Form fields 
+1. The request body (For [controllers that have the [ApiController] attribute](xref:web-api/index#binding-source-parameter-inference).)
+1. Route data
+1. Query string parameters
+1. Uploaded files 
+
+For each target parameter or property, the sources are scanned in the order indicated in this list. There are a few exceptions:
+
+* Form fields and request body data are used only for properties of complex types.
+* Route data and query string values are used only for simple types.
+* Uploaded files are bound only to target types that implement `IFormFile` or `IEnumerable<IFormFile>`.
+
+If the default behavior doesn't give the right results, you can use one of the following attributes to specify the source to use for any given target. 
+
+* `[FromHeader]` attribute
+* `[FromQuery]` attribute
+* `[FromRoute]` attribute
+* `[FromForm]` attribute
+* `[FromBody]` attribute
 
-Note: The strings in the URL route are not case sensitive.
+### [FromHeader] attribute
 
-MVC will try to bind request data to the action parameters by name. MVC will look for values for each parameter using the parameter name and the names of its public settable properties. In the above example, the only action parameter is named `id`, which MVC binds to the value with the same name in the route values. In addition to route values MVC will bind data from various parts of the request and it does so in a set order. Below is a list of the data sources in the order that model binding looks through them:
+Gets values from HTTP headers. For some headers, you can't make a C# parameter or property name that matches the header name because the header name has a hyphen in it.  In these cases, provide the header name to the attribute constructor:
```

Comment:
> For some headers, you can't make a C# parameter or property name that matches the header name because the header name has a hyphen in it.  In these cases, provide the header name to the attribute constructor:

Should we make this a general purpose comment for the `From*` attribute list above? It's just as valid for query string parameters as it is for headers. Something along the lines of:

> All of these attributes optionally accept a model name value in the constructor. This is provided in case the property name does not match the value in the Request or if the value in the Request cannot be represented as a C# property (for instance, a header name might have a hypen in it).